### PR TITLE
Generic deployment reconciler

### PR DESCRIPTION
**Pull request by roivaz**
Create a generic Deployment reconciler that can be used from several controllers. Also fix a bug detected in the reconciliation of the MutatingWebhookConfiguration.

### CODE REVIEW

File: pkg/reconcilers/deployment.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+			r.logger.Info("Created Deployment")
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	var updated bool = false
+	desired := generatorFn()
+
+	tmpUpdated, err := r.reconcileSpec(deployment, desired)
```
Comment (raelga) [inline pkg/reconcilers/deployment.go:70]: What do you thing about wrapping the `r.reconcileSpec` and `r.reconcileLabels` in a `r.reconcileDeployment`, so in the Reconcile we have a single call returning the updated boolean and the errors?

File: pkg/controller/discoveryservice/deployment.go [hunk 2]
```diff
@@ -1,196 +1,154 @@
...
 
-	if r.ds.Spec.Debug {
-		dep.Spec.Template.Spec.Containers[0].Args = append(dep.Spec.Template.Spec.Containers[0].Args, "--zap-devel")
-	}
+		if ds.Spec.Debug {
+			dep.Spec.Template.Spec.Containers[0].Args = append(dep.Spec.Template.Spec.Containers[0].Args, "--zap-devel")
+		}
 
-	return dep
+		return dep
```
Comment (raelga) [inline pkg/controller/discoveryservice/deployment.go:152]: ✨

**Comment by roivaz** [comment]
@raelga I have joined in both the spec and labels reconciliation in a single `reconcileDeployment` function. I have also added tests.

**Comment by raelga** [comment]
Awesome @roivaz, re-approved.

**Pull request merged**